### PR TITLE
disallow miners from not serving axons to keep prior incentive from previous miner

### PR DESCRIPTION
**Pull request by ifrit98**
Allows miners to be "available" to the validator even if not publishing an axon so that miners will fail and be unable to farm previous registered incentive.

### CODE REVIEW

**Review by mjurbanski-reef — APPROVED** [review]
I guess an argument to do it this way (instead of just during weight distribution) is since it should addresses potential problem with down detection in monitor function https://github.com/ifrit98/storage-subnet/blob/def46f8483b8bdeb4c83f38fe92590911c7866e3/storage/validator/network.py#L190

File: storage/validator/utils.py [hunk 1]
```diff
@@ -132,13 +132,13 @@ def check_uid_availability(
...
-    # Filter non serving axons.
-    if not metagraph.axons[uid].is_serving:
-        return False
     # Filter validator permit > 1024 stake.
     if metagraph.validator_permit[uid]:
         if metagraph.S[uid] > vpermit_tao_limit:
             return False
+    # Disallow non-serving miner axons by letting them fail early. (Excludes validators)
+    if metagraph.axons[uid].ip == '0.0.0.0':
+        return True
```
Comment (mjurbanski-reef) [inline storage/validator/utils.py:141]: at this point this line is just doing nothing, i.e. could be
```suggestion
```

Comment (mjurbanski-reef) [reply storage/validator/utils.py:141]: the docstring should be updated instead, but I guess that can wait after Hotfix:)

Comment (ifrit98) [reply storage/validator/utils.py:141]: Later

Comment (ifrit98) [reply storage/validator/utils.py:141]: Family time now :)

**Pull request merged**